patches/tModLoader/Terraria/ModLoader/ModType.cs

```diff
@@ -0,0 +1,34 @@
+﻿namespace Terraria.ModLoader
+{
+	/// <summary>
+	/// The base type for most modded things.
+	/// </summary>
+	public abstract class ModType : IModType
+	{
+		///<summary>
+		/// The mod this belongs to.
+		/// </summary>
+		public Mod Mod { get; internal set; }
+
+		/// <summary>
+		/// The internal name of this.
+		/// </summary>
+		public virtual string Name => GetType().Name;
+
+		void ILoadable.Load(Mod mod) {
+			Mod = mod;
+			Load();
+			AddInstance();
+		}
+
+		public virtual void Load(){}
+
+		public virtual void Unload(){}
+
+		/// <summary>
+		/// DO NOT CALL THIS! This is called automatically in ILoadable.Load
+		/// </summary>
+		internal virtual void AddInstance(){}
```

Comment:
shouldn't this be protected?

Reply:
I admit I went back and forth on this method entirely. In a perfect world, the new Load() method handle it and then anyone wanting to override it would call base.Load(). Given that most tML hooks don't require this, I decided to have a separate method for this with the build-in types to ensure that they get loaded. Anyone wanting to extend this base class can simply just add their stuff in the Load method. By making it internal, no modders can accidentally call it or replace it.